Mining Textures & More: Revival

Sun-Soaked opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
- A fixed version of the PR https://github.com/shiptest-ss13/Shiptest/pull/429, containing all it's content wholesale, albeit with some minor pathing adjustments to hopefully fix previous merge conflicts.
- Also includes the previous mining suit reskin as a new mining suit subtype.
- Gives holoparasites very slightly increased damage absorption and the ability to fly in zero g.

I'll work on some implementation stuff when the research PR is merged and I can touch maps again.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Fixes a number of bugs I accidentally caused with previous mining PRs
Adds new and purty mining textures, as well as a bit of setting fleshing and some new loot.
Jackhammers now function as demolition tools as well as mining ones, mimicking how they're used in real life.

## Changelog
:cl:
add: The Syndicate Cleaver, a syndiepilled crusher option. You've never seen anything like it before. Ever. Don't make them sue.
add: legacy mining equipment, slightly more powerful versions of typical exploration gear that also cause mild slowdown.
add: The Heavy Mining Suit, for heavy-duty resource extraction jobs.
add: new multi-functional equipment for the syndicate Brigador cyborg, as well as a new sprite.
add: varied references to EXOCON, a new-age frontier operations corp.
add: jackhammer wall and window smashing functionality.
add: an exocon-themed secure crate.
add: some showcases, to be used in a mapping project.
tweak: set up a few new ss13 archetype corpses, usable in mapping
tweak: added some backpack loot to legion miner drops, and an Old Miner corpse drop.
balance: the legion staff now fires significantly weaker skulls.
balance: boosts the power of several holoparasites, gives holoparasites spacewalking.
fix: replaced ice demon crusher loot drops to prevent runtimes
fix: fixed a continuous clown simplemob-related runtime
imageadd: added a number of new mining-related sprites and recolors
imageadd: ported resprites of the singularity hammer and mjolnir
imageadd: added marg.
imageadd: an ominous wired-up brigador. Where could it possibly be used? find out on the next episode of dragon shiptest z
spellcheck: fixed some minor typos and spacing issues here and there
add: marg.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Sun-Soaked committed:
the iceman cometh

-adds all content from previous PR
-I cannot test this currently because VSC is broken, so prepare for a merge conflict

Sun-Soaked committed:
rock taste like home

-rock eater get new suit

Sun-Soaked commented:
> albeit with some minor pathing adjustments to hopefully fix previous merge conflicts.

aaaa

Sun-Soaked committed:
fixes 01

right, let's see if this helps

Sun-Soaked committed:
fix 02

alright that one's my bad

Sun-Soaked committed:
fixes 3

-implements syndie crushes and brigador temporarily, so we can get field testing

-fixes missing (loc) issues

Sun-Soaked committed:
marg.

emergency margfix

Sun-Soaked committed:
g

m

Apogee-dev commented:
Considering the introduction of a different mining hardsuit in #455, perhaps it might make more sense to flavor the original mining hardsuit as Nanotrasen-issue, since the intent is to replace them on most non-nanotrasen ships anyway? Or to otherwise figure out how we're going to place these three hardsuits relative to each other, I suppose.

Sun-Soaked commented:
I worked about with this for a while, and I feel like the Heavy Mining suit is ultimately intended to be a premium mining suit you buy ig through cargo/it's appends, as opposed to a starter suit. 
I'll implement a cargo buy for it, and buff it slightly.

Apogee-dev commented:
> I worked about with this for a while, and I feel like the Heavy Mining suit is ultimately intended to be a premium mining suit you buy ig through cargo/it's appends, as opposed to a starter suit. I'll implement a cargo buy for it, and buff it slightly.

Yeah, I got that impression about the heavy suit through its description and stats. I was mostly concerned about the standard mining hardsuit being reworked to be more of a generic mass-produced one rather than a Nanotrasen special, since that's the niche the one in the indie suit bundle is intended to fill.

Sun-Soaked committed:
wee

-adds a heavy mining cargo kit
-fixes
-adds a slightly less powerful stimpen

Sun-Soaked commented:
if I'm being honest, the basic suit could use a rework to look more like an NT design, as well. Currently it feels like something people just kind of wear, like the t-shirt of the frontier- it's also kind of just a really noisy sprite without much actually going on.
Ideally it would look more like the Old Exploration Gear in this pr, in terms of flavour and color.

Sun-Soaked commented:
![image](https://user-images.githubusercontent.com/45698967/143370917-15f5e728-40d3-4521-8b20-da548e808c58.png)
(some examples of that)

Apogee-dev commented:
> if I'm being honest, the basic suit could use a rework to look more like an NT design, as well. Currently it feels like something people just kind of wear, like the t-shirt of the frontier- it's also kind of just a really noisy sprite without much actually going on. Ideally it would look more like the Old Exploration Gear in this pr, in terms of flavour and color.

I think it does effectively accomplish the original idea behind it- which is to say, it's a Dead Space rigsuit. They're kind of noisy by nature, and redesigning it would lose the instant recognizability of the "classic" miner hardsuit that most other codebases have. At the moment I do think the nanotrasen hardsuits, by and large, are pretty recognizable and have some common features, like the triple vision slits, that make them recognizably Nanotrasen. The only real exception to that that isn't normally a unique item is the medic hardsuit, which frankly does need a redesign to better fit with the other NT suits.

Sun-Soaked committed:
g

-slightly nerfs hvymining in light of being able to reinforce it

triplezeta commented:
ill resprite the mining hardsuit as faithfully as i can at a later date

triplezeta commented:
along with every other hardsuit most of them kind of smell

Apogee-dev commented:
> along with every other hardsuit most of them kind of smell

They could do with a little more clarity. The engineering hardsuit is the best of them, i really dislike the sec suit and prefer the design used on vg (which isn't just native server bias, the palette's a lot better and it's easier to read the suit's shape)

Sun-Soaked committed:
testmerge sus

-I can see you're: Merging multiple conflicting prs. Would you like help with that?

Sun-Soaked committed:
why

Sun-Soaked committed:
fine, let's play tough

a

Sun-Soaked committed:
Merge branch 'master' into miningtextures

Sun-Soaked commented:
![image](https://user-images.githubusercontent.com/45698967/143379963-eefefc8c-b280-40d4-bbb6-8771bc5498ba.png)
what did they mean by this

triplezeta commented:
![image](https://user-images.githubusercontent.com/51932018/143380038-7c969a0d-4dfb-4dc0-8ed2-ec831b291f62.png)
it means this

Sun-Soaked committed:
guh

-interesting

Sun-Soaked committed:
tiny baby man

-fix

Sun-Soaked committed:
AAAAAAAAAAAA

MENACCINGMENACINGMENACINGMENACINGMENACNG

Sun-Soaked committed:
the hammer comes down

-.38 hunting now does 50 damage to simplemobs instead of 75(still really high but less completely stupid)

Sun-Soaked commented:
please father,  I don't want to manually remove and re-import all my icons again

Sun-Soaked committed:
royal flush pt. 1

now's the time to play your hand

Sun-Soaked committed:
burghley

-no more spunge bob.
(if this does not work I will commit acts of great and terrible seething)

Sun-Soaked commented:
HOW

Sun-Soaked committed:
holotweeks

-adjusts the balance of a number of holoparasites, making them more viable in shiptests gun/sword heavy gameplay enviroment
-allows holoparasites to fly freely in zero-grav

Fikou commented on code/game/objects/structures/window.dm:
```diff
@@ -190,6 +190,16 @@
+	if(istype(I, /obj/item/pickaxe/drill/jackhammer))
+		to_chat(user, "<span class='notice'>You begin to smash though [src]...</span>")
+		if(do_after(user, 75, target = src))
+			if(!istype(src, /obj/structure/window))
+				return TRUE
```
jesse

Sun-Soaked replied:
This is behavior inherited from other jackhammer smash procs I ported from tg, you'll see they have a similar define. I think it prevents a bug related to triggering the effect on other objects after clicking a window?
Not entirely sure, but I made sure to preserve it.

Sun-Soaked committed:
Merge branch 'master' into miningtextures

Sun-Soaked commented:
wooo yeah woooo yeah woooo yeah

Sun-Soaked committed:
fixes

-fixes legates dying on spawn

Sun-Soaked committed:
m

m

Sun-Soaked committed:
the great backpackification

-nerfs some backpack probs, reduces the chance of the better soldier appearing, adds mesons to the miner corpse, adds backpacks to several other legion spawns

Sun-Soaked committed:
gack gack gack

-makes necroseed slightly harder to get, in light of how powerful it is when a viro aquires it

Sun-Soaked committed:
spam moment

Sun-Soaked committed:
buffs

-slightly buffs two elite items
-increases chance of tumour shard drop chance, allows it to drop alongside elite loot when pvp is done

Sun-Soaked committed:
:mothdonk:

undocumented, unfinished changes

Sun-Soaked committed:
crazy noisy bizzare fix

-white space

Apogee-dev commented:
A small request, but could we get a legion corpse type with the asteroid miner jumpsuit + indie mining hardsuit? Wanted to add that to my own PR before it got merged but i honestly expected this one to be merged first.

Sun-Soaked committed:
Merge branch 'master' into miningtextures

Sun-Soaked committed:
protosuit buff

guh

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into miningtextures

Sun-Soaked commented:
:videogames:
